utility/service/service.go

```diff
@@ -1,11 +1,217 @@
+	base_modules.InterruptableModule
+
+	logger *modules.Logger
+	config *configs.ServicerConfig
+}
+
+func CreateServicer(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
+	return new(servicer).Create(bus, options...)
+}
+
+func (*servicer) Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
+	s := &servicer{
+		logger: logger.Global.CreateLoggerForModule(servicerModuleName),
+	}
+
+	for _, option := range options {
+		option(s)
+	}
+
+	bus.RegisterModule(s)
+
+	cfg := bus.GetRuntimeMgr().GetConfig()
+	s.config = cfg.Utility.ServicerConfig
+
+	return s, nil
+}
+
+func (s *servicer) Start() error {
+	s.logger = logger.Global.CreateLoggerForModule(s.GetModuleName())
+	return nil
+}
+
+func (*servicer) GetModuleName() string {
+	return servicerModuleName
+}
+
+// HandleRelay processes a relay after performing validation.
+// It also updates the servicer's internal state to keep track of served relays.
+func (s *servicer) HandleRelay(relay *coreTypes.Relay) (*coreTypes.RelayResponse, error) {
+	if relay == nil {
+		return nil, fmt.Errorf("cannot serve nil relay")
+	}
+
+	if err := s.admitRelay(relay); err != nil {
+		return nil, fmt.Errorf("Error admitting relay: %w", err)
+	}
+
+	// TODO: implement Persist Relay
+	// TODO: implement execution
+	// TODO: implement state maintenance
+	// TODO: validate the response from the node?
+	// TODO: (QUESTION) Should we persist SignedRPC?
+	return nil, nil
+}
+
+// validateRelayMeta ensures the relay metadata is valid for being handled by the servicer
+// TODO: (REFACTOR) move the meta-specific validation to a Validator method on RelayMeta struct
+func (s servicer) validateRelayMeta(meta *coreTypes.RelayMeta, currentHeight int64) error {
+	if meta == nil {
+		return fmt.Errorf("empty relay metadata")
+	}
+
+	if meta.RelayChain == nil {
+		return fmt.Errorf("relay chain unspecified")
+	}
+
+	// TODO: supported chains: needs to be crossed-checked with the world state from the persistence layer
+	if err := s.validateRelayChainSupport(meta.RelayChain, currentHeight); err != nil {
+		return fmt.Errorf("validation of support for relay chain %s failed: %w", meta.RelayChain.Id, err)
+	}
+
+	return nil
+}
+
+func (s servicer) validateRelayChainSupport(relayChain *coreTypes.Identifiable, currentHeight int64) error {
+	if !isChainSupported(relayChain.Id, s.config.Chains) {
+		return fmt.Errorf("chain %s not supported by servicer %s configuration", relayChain.Id, s.config.Address)
+	}
+
+	// DISCUSS: either update NewReadContext to take a uint64, or the GetCurrentHeight to return an int64.
+	readCtx, err := s.GetBus().GetPersistenceModule().NewReadContext(currentHeight)
+	if err != nil {
+		return fmt.Errorf("error getting persistence context at height %d: %w", currentHeight, err)
+	}
+	defer readCtx.Release() //nolint:errcheck // We only need to make sure the readCtx is released
+
+	// DISCUSS: should we update the GetServicer signature to take a string instead?
```

Comment:
Yes

![Screenshot 2023-05-31 at 7 16 11 PM](https://github.com/pokt-network/pocket/assets/1892194/47b47354-85ad-426b-95e1-83073bae38d5)

Reply:
There might be an issue here when we connect everything (i.e. needing to do hex decoding/encoding) but it's something we'll simplify later and hit an error on when doing the business logic anyhow.

Reply:
Thanks, seem like the comment can be kept then.